eth/filters/api.go

```diff
@@ -330,15 +330,13 @@ func (api *PublicFilterAPI) GetLogs(ctx context.Context, crit FilterCriteria) ([
 		filter = NewBlockFilter(api.backend, *crit.BlockHash, crit.Addresses, crit.Topics)
 	} else {
 		// Convert the RPC block numbers into internal representations
-		var (
-			begin int64
-			end   int64
-		)
-		if crit.FromBlock == nil {
-			begin = int64(rpc.LatestBlockNumber)
+		begin := rpc.LatestBlockNumber.Int64()
```

Comment:
nit: seems a bit wasteful to first get the value and then don't use it in some cases. How about `if crit.FromBlock != nil begin = crit.FromBlock.Int64() else begin = rpc.LatestBlockNumber.Int64()`

SCNR: would be even nicer in a language that supports this: begin = if (crit.FromBlock != nil) crit.FromBlock.Int64() else  rpc.LatestBlockNumber.Int64()`

Reply:
LatestBlockNumber is just a big.Int constant, so the cost of this op is a cast from big.Int to int64. The more optimal version isn't really worth the more complex code.

Reply:
I do not think the code is more complex this way. I think this way it is more expressive and tells the developer what it is doing instantly - saving a few cycles in the process. But might also just be my language bias as I would have this in mind but it is not possible in go AFAIK: `begin = (crit.FromBlock?:rpc.LatestBlockNumber).Int64`
As I prefixed - just a nit & just something that went through my mind when reading the code - no real need to change something.